doc/rank_evaluation.md

```diff
@@ -0,0 +1,131 @@
+# Rank evaluation
+
+One way we assess the affect of search-api changes to relevancy, is by using
+Elasticsearch's [Ranking Evaluation API](ranking_evaluation_api).
+
+The API enables us to score how well search-api ranks results for a given query.
+
+## What is rank evaluation?
+
+Rank evaluation is a process by which we rate how well search-api ranks results for
+a given query.
+
+For example, given the query `Harry Potter` in an ideal situation, the results
+would be returned in this order:
+
+1. Harry Potter
+2. Harry Potter World
+3. Who is Harry Potter?
+
+However, the results might come back in this order:
+
+1. Who is Harry Potter?
+2. Sign in to your HMRC account
+3. Harry Potter
+
+The first situation is good and the second is bad. But what we really
+want is a metric to say *how bad* the results are.
+
+Rank evaluation provides us with a metric which tells us how good or bad
+the results are for a set of queries that we have already manually ranked.
+
+So we have a rake task `debug:rank_evaluation` that does this.
+
+This is useful, because we can tell how good a change is before running an
+AB test on real users.
+
+For example:
+
+```
+$ rake debug:rank_evaluation
+
+harry potter:                      0.6297902553883483
+passport:                          0.7926871059630242
+contact:                           0.9957591943893841
+...
+Overall score:                     0.8209391842392532 (average of all scores)
+```
+
+The above means that the queries for `passport` and `contact` are both
+returning better results than the query for `harry potter`.
+
+Moreover, if the score for `harry potter` was `0.2392687105963024` before
+we made a change, then that means we've made a good change for that query.
+
+This can be measured over time, and it is! See the section on 'What do we do
+with the query scores?' below.
+
+## How do we compute a score for a query?
+
+Given a set of queries and a list of manually rated documents, the API tells
+us how well we are ranking the results of queries given the manual ratings
+we have supplied.
+
+A score of 1 is perfect and a score of 0 is catastrophic.
+
+But how do we get to the number `0.6297902553883483` for the query `harry potter`?
+
+We use [nDCG](ncdg) (normalised Discounted Cumulative Gain). DCG is a measure of ranking quality.
+
+From Wikipedia:
+
+By using DCG we make two assumptions:
+
+> - Highly relevant documents are more useful when appearing earlier in a search engine result list (have higher ranks)
+> - Highly relevant documents are more useful than marginally relevant documents, which are in turn more useful than non-relevant documents.
+
+DCG requires a query and a list of relevancy judgements (rated documents).
+
+We manually give a rating between 0 and 4 to documents in search results.
+4 = Totally relevant, 0 being not relevant.
```

Comment:
It's actually 0 to 3 in our current judgements:

- 0 = irrelevant (which is equivalent to unrated in DCG)
- 1 = misplaced
- 2 = near
- 3 = relevant

Reply:
Thanks @barrucadu, I added this to the doc!